tests/test_lazyflow/test_operators/test_ioOperators/test_pixel_sizes.py

```diff
@@ -423,3 +432,246 @@ def test_write_read_roundtrip_h5(
     for axis in axes:
         assert meta.axistags[axis].resolution == resolutions[axes.index(axis)]
         assert meta.axis_units[axis] == units[axes.index(axis)]
+
+
+def setup_op_data_lane(dataset_infos: List[FilesystemDatasetInfo], graph):
+    roles = ["Raw Data", "Secondary", "Tertiary"]
+    assert len(roles) == len(dataset_infos), "bad setup"
+    op_data = OpDataSelectionGroup(graph=graph)
+    op_data.WorkingDirectory.setValue(os.getcwd())
+    op_data.DatasetRoles.setValue(roles)
+    op_data.DatasetGroup.setValues(dataset_infos)
+    return op_data
+
+
+def assert_eq_spatial_pixel_size(actual_slot: Slot, expect_tags: vigra.AxisTags, expect_units: Dict[str, str]):
+    for actual_tag in actual_slot.meta.axistags:
+        if actual_tag.key == "c":
+            continue
+        assert actual_tag in expect_tags, f"axis {actual_tag.key} was not expected"
+        assert np.isclose(
+            actual_tag.resolution, expect_tags[actual_tag.key].resolution, atol=1e-15
+        ), "slot axis resolution: actual != expected"
+    assert actual_slot.meta.axis_units == expect_units, "actual slot axis units != expected"
+
+
+@pytest.fixture
+def dataset_with_pixel_size(inputdata_dir) -> FilesystemDatasetInfo:
+    return FilesystemDatasetInfo(filePath=inputdata_dir + "/pix_res/3d_t.tif")
+
+
+@pytest.fixture
+def dataset_no_pixel_size(inputdata_dir) -> FilesystemDatasetInfo:
+    info = FilesystemDatasetInfo(filePath=inputdata_dir + "/pix_res/3d_t.tif", axistags=vigra.defaultAxistags("tzyx"))
+    info.axis_units = None
+    return info
+
+
+@pytest.fixture
+def dataset_other_pixel_size(inputdata_dir) -> FilesystemDatasetInfo:
+    info = FilesystemDatasetInfo(filePath=inputdata_dir + "/pix_res/3d_t.tif")
+    for tag in info.axistags:
+        info.axistags.setResolution(tag.key, 3.1415)
+    info.axis_units = {"t": "sec", "x": "nm", "y": "cm", "z": "micron"}
+    return info
+
+
+def test_DataSelection_roles_copies_from_raw_data(graph, dataset_with_pixel_size, dataset_no_pixel_size):
+    dataset_infos = [dataset_with_pixel_size, dataset_no_pixel_size, dataset_no_pixel_size]
+    op_data = setup_op_data_lane(dataset_infos, graph)
+
+    for i in range(3):
+        assert_eq_spatial_pixel_size(
+            op_data.ImageGroup[i], dataset_with_pixel_size.axistags, dataset_with_pixel_size.axis_units
+        )
+    assert OpDataSelectionGroup.META_COPY_KEY in op_data.ImageGroup[1].meta
+    assert op_data.ImageGroup[1].meta[OpDataSelectionGroup.META_COPY_KEY] == "Raw Data"
+    assert OpDataSelectionGroup.META_COPY_KEY in op_data.ImageGroup[2].meta
+    assert op_data.ImageGroup[2].meta[OpDataSelectionGroup.META_COPY_KEY] == "Raw Data"
+
+
+def test_DataSelection_roles_copies_from_raw_data_2roles(graph, dataset_with_pixel_size, dataset_no_pixel_size):
```

Comment:
```suggestion
def test_DataSelection_roles_copies_from_raw_data_2(graph, dataset_with_pixel_size, dataset_no_pixel_size):
```
probably accidental ctrl+v?

Reply:
ah right - yeah I think any of your suggestions will make it easier to understand!